File: qr-check-latestcode/QR_CHECK_IN/code/app/src/main/java/com/example/qr_check_in/Notification/RetrofitInstance.java
package com.example.qr_check_in.Notification;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * retorifit instance to call API
 */

public class RetrofitInstance {
    private static final String BASE_URL = "https://fcm.googleapis.com";
    private static Retrofit retrofit = null;
    private static NotificationAPI api = null;

    static {
        retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        api = retrofit.create(NotificationAPI.class);
    }

    public static NotificationAPI getApi() {
        return api;
    }
}

File: Listerpros/app/src/main/java/com/example/listerpros/api/RetrofitInstance.kt
package com.example.listerpros.api

import com.example.listerpros.constants.Constants.Companion.BASE_URL
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit


object RetrofitInstance {



    private val retrofit by lazy {
        val gson: Gson = GsonBuilder().setLenient().create()
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create(gson))
            .build()
    }

    val api: Api by lazy {
        retrofit.create(Api::class.java)
    }
}